java/ql/src/Telemetry/AutomodelSharedCharacteristics.qll

```diff
@@ -0,0 +1,281 @@
+float maximalConfidence() { result = 1.0 }
+
+float highConfidence() { result = 0.9 }
+
+float mediumConfidence() { result = 0.6 }
+
+/**
+ * A specification of how to  instantiate the shared characteristics for a given candidate class.
+ *
+ * The `CandidateSig` implementation specifies a type to use for Endpoints (eg., `ParameterNode`), as well as a type
+ * to label endpoint classes (the `EndpointType`). One of the endpoint classes needs to be a 'negative' class, meaning
+ *   "not any of the other known endpoint types".
+ */
+signature module CandidateSig {
+  class Endpoint;
+
+  class EndpointType;
+
+  /** Gets the string representing the file+range of the endpoint. */
+  string getLocationString(Endpoint e);
```

Comment:
Will this ever be something other than `e.getLocation().toString()`? If not, I suggest omitting it from the signature, and just writing the default implementation.

Reply:
It always would've been `e.getLocation().toString()` or something very similar. I've managed to remove the predicate altogether, which (I think) you will like even better.